### DISCUSSION TITLE
Why is the deepmd-kit so fast?

### DISCUSSION THREAD

**Discussion by kudoryutaro** [deepmodeling/deepmd-kit#2262] [General]
Hello.

I'm making DeepPot-SE AtomTypeEmbedding version using pytorch.

So my question is, why is deepmd-kit so fast?

I implemented the forward and backward of the R_i -> R^Tilda_i part by hand using pytorch. 

However, it takes 0.1 sec to infer the force and potential energy with 5000 atoms at a time. 

On the other hand, deepMD-kit can calculate about 40000 atoms in 0.1 sec. Why is it so fast?

**Comment by wanghan-iapcm** [comment]
Hello,

I cannot tell you the different unless we have detailed benchmarks for both implementations.  One possibility is the difference of deep learning platforms. DeePMD-kit uses tensorflow v1 which uses static graph, while pytorch uses dynamic graph.
